Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+           $logResource.Add("Reason","Minimum TLS Version is set on all the custom domains of front door.")    
+           $logSkippedResources += $logResource
+        }
+        else 
+        {
+            if (-not $frontDoorsEndpointWithoutReqMinTLSVersion.MinimumTlsVersion -ne $requiredMinTLSVersion)
+            {
+                Write-Host "Minimum TLS Version is not set on the custom domain"  -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            $NonCompliantFrontDoorEndpoints += $frontDoorsEndpointWithoutReqMinTLSVersion | Select-Object @{N='ResourceGroupName';E={$frontDoorsEndpointWithoutReqMinTLSVersion.ResourceGroupName}},
+                                                                                            @{N='ResourceName';E={$frontDoorsEndpointWithoutReqMinTLSVersion.ResourceName}},
+                                                                                            @{N='DomainName';E={$frontDoorsEndpointWithoutReqMinTLSVersion.DomainName}},
+                                                                                            @{N='CertificateType';E={$frontDoorsEndpointWithoutReqMinTLSVersion.CertificateType}},
+                                                                                            @{N='MinimumTlsVersion';E={$frontDoorsEndpointWithoutReqMinTLSVersion.MinimumTlsVersion}},
+                                                                                            @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorsEndpointWithoutReqMinTLSVersion.isMinTLSVersionSetOnCustomDomain}}
+        }
+    }
+    
+    $TotalNonCompliantFrontDoorEndpoints = ($NonCompliantFrontDoorEndpoints | Measure-Object).Count
+
+    if ($totalNonCompliantFrontDoorEndpoints -eq 0)
+    {
+        Write-Host "No Front Door(s) found having minimum TLS version less than required minimum TLS version. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation -and ($frontDoorsWithReqMinTLSVersion|Measure-Object).Count -gt 0) 
+        {
+            $logFile = "LogFiles\"+ $($TimeStamp) + "\log_" + $($SubscriptionId) +".json"
+            $log =  Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach($logControl in $log.ControlList){
+                if($logControl.ControlId -eq $controlIds)
+                {
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalNonCompliantFrontDoorEndpoints)] out of [$($totalFrontDoors)] Front Door(s) having minimum TLS version less than required minimum TLS version." -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
$totalNonCompliantFrontDoorEndpoints, $totalFrontDoors 
Does both of these collection contains front doors from name I can see first one contains endpoint while other front doors?

Reply:
Yes Tarun first contains endpoints and other contain total frontdoors. It seems confusing so I have changed it to "Found [$($totalNonCompliantFrontDoorEndpoints)] Endpoints of [$($totalFrontDoors)] Front Door(s) having minimum TLS version less than required minimum TLS version."